app/controllers/account_requests_controller.rb

```diff
@@ -11,8 +11,7 @@ def received; end
   def confirmation; end
 
   def confirm
-    @account_request.update!(confirmed_at: Time.current)
-    AccountRequestMailer.approval_request(account_request_id: @account_request.id).deliver_later
+    @account_request.approve!
```

Comment:
I think this should be actually the step before `approval`. This endpoint was handling the prospective user's confirmation that they do want the account. Approve comes from admins.

Edit: Just saw the definition of `approve!` could modify the name of the method to avoid confusion?